File: node-http/models/friendRelation.js
/**
 * @Author:acexy@thankjava.com
 * 18/8/20
 * @Description:
 */
const baseModel = require('./basic.js');

class friendRelation extends baseModel {

    Schema() {
        return {
            userId: {type: String, required: true},
            friendUserId: {type: String, required: true},
            remark: {type: String, required: false},
        };
    };

    SchemaName() {
        return 'friend_relation';
    };


    selectByUserId(userId) {
        return this.model.find({userId: userId}).exec();
    }

    selectByConditionOnlyOne(friendRelation) {
        return this.model.findOne(friendRelation).exec();
    }

    updateById(id, friendRelation) {
        return this.model.updateOne({_id: baseModel.typeObject(id)}, friendRelation).exec();
    }
}

module.exports = new friendRelation();

File: java-websocket/src/main/java/com/thankjava/wchat/db/entity/FriendRelation.java
package com.thankjava.wchat.db.entity;

/**
 * @Author: acexy@thankjava.com
 * 2018/8/14
 * @Description: 好友关系中间信息
 **/
public class FriendRelation extends BaseEntity {

    private String userId;
    private String friendUserId;
    private String remark;

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getFriendUserId() {
        return friendUserId;
    }

    public void setFriendUserId(String friendUserId) {
        this.friendUserId = friendUserId;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

}
